File: Royal-Cyber-iOS/Post.swift
import Foundation

class Post {
    var imageUrl: String
    var caption: String

    init(imageUrl: String, caption: String) {
        self.imageUrl = imageUrl
        self.caption = caption
    }

    func getImageUrl() -> String {
        return imageUrl
    }

    func setImageUrl(imageUrl: String) {
        self.imageUrl = imageUrl
    }

    func getCaption() -> String {
        return caption
    }

    func setCaption(caption: String) {
        self.caption = caption
    }
}

File: app/src/main/java/com/example/instagram/Post.java
package com.example.instagram;

public class Post {
    private String imageUrl;
    private String caption;
    private String location;
    private long scheduledTime; // New attribute to store the scheduled time

    public Post(String imageUrl, String caption, String location) {
        this.imageUrl = imageUrl;
        this.caption = caption;
        this.location = location;
    }

    public String getImageUrl() {
        return imageUrl;
    }

    public void setImageUrl(String imageUrl) {
        this.imageUrl = imageUrl;
    }

    public String getCaption() {
        return caption;
    }

    public void setCaption(String caption) {
        this.caption = caption;
    }

    public String getLocation() {
        return location;
    }

    public void setLocation(String location) {
        this.location = location;
    }

    public long getScheduledTime() { // Getter method for scheduledTime
        return scheduledTime;
    }

    public void setScheduledTime(long scheduledTime) { // Setter method for scheduledTime
        this.scheduledTime = scheduledTime;
    }
}
